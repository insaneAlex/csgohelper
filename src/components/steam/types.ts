import {NumObject} from '@/types';

export enum PriceOptions {
  DAY = '24_hours',
  WEEK = '7_days',
  MONTH = '30_days',
  ALL = 'all_time'
}

export type GridConfigType = Record<string, NumObject>;
