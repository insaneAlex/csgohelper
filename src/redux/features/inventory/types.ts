import {SteamIDType} from '../../types';

export enum InventoryStatuses {
  IDLE = 'idle',
  INIT_LOAD = 'Init loading',
  FORCE_LOAD = 'Force loading',
  NO_PROFILE = 'Request failed with status 404',
  PRIVATE_INVENTORY = 'Request failed with status 403'
}

export type GetInventoryPayloadType = SteamIDType & {isForceUpdate?: boolean};
