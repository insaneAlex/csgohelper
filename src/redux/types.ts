import {InventoryItemType} from '@/types';
import {CS2InventoryFetchErrorType} from '@/pages/api/csgoInventory';

export type SteamIDType = {steamid: string};
export type InventoryErrorType = null | string | CS2InventoryFetchErrorType;

export type InventoryState = {
  isLoading: boolean;
  error: InventoryErrorType;
  items: InventoryItemType[];
  update_time?: string | null;
};
