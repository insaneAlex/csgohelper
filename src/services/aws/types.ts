import {SteamProfileType} from '@/core/types';
import {InventoryItemType} from '..';
import {PriceType} from '../types';

export type AWSConfigType = {region: string; credentials: {accessKeyId: string; secretAccessKey: string}};

export type AmazonResponseType = {
  '$metadata': {
    httpStatusCode: number;
    requestId: string;
    extendedRequestId?: string;
    cfId?: string;
    attempts: number;
    totalRetryDelay: number;
  };
};

export type PricesType = Record<string, {price: PriceType}> | null;

export type NoPriceInventory = Omit<InventoryItemType, 'prices'>[];
export type InventoryRecordType = {inventory?: NoPriceInventory; update_time?: string; profile?: SteamProfileType};
export type InventoryCacheType = Record<string, InventoryRecordType>;
