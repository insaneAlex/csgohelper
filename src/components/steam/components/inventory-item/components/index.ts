export {ItemOpened} from './item-opened';
