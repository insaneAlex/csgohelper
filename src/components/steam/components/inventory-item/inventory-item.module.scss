@import 'styles/variables.scss';

.item {
  padding: 5px;
  position: relative;
  width: calc(100% / 9);
  transition: width 100ms;

  @media (max-width: $xx-large) {
    width: calc(100% / 6);
  }
  @media (max-width: $large) {
    width: calc(100% / 4);
  }
  @media (max-width: $medium) {
    width: calc(100% / 3);
  }
  @media (max-width: $mobile) {
    width: calc(100% / 2);
  }
}

.link {
  border-radius: 3px;
  display: grid;
  height: 125px;
  padding: 10px;
  place-items: center;
  @include body-short-link;
}

.price {
  left: 0;
  padding: 15px;
  position: absolute;
  top: 0;
}

.describe {
  font-size: 13px;
  text-align: center;
  white-space: nowrap;
  overflow: hidden;
  width: 100%;
  text-overflow: ellipsis;
  @media (max-width: $large) {
    font-size: 12px;
  }
  @media (max-width: $mobile) {
    font-size: 10px;
  }
}
