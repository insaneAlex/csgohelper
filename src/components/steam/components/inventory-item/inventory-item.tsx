import {inventoryImageBaseUrl} from '../../constants';
import {ImgSize, InventoryItemType} from '@/types';
import {STAT_TRAK_PATTERN} from './constants';
import {Layout} from 'react-grid-layout';
import {PriceOptions} from '../../types';
import Image from 'next/image';
import Link from 'next/link';
import {FC} from 'react';

import styles from './inventory-item.module.scss';

type Props = {imgSize: ImgSize; item: InventoryItemType & Layout};

export const InventoryItem: FC<Props> = ({item, imgSize}) => {
  const {name_color, assetid, icon_url, count, prices} = item;
  const {width, height} = imgSize;

  const price = prices?.[PriceOptions.MONTH]?.average * (count || 1);
  const formattedPrice = !isNaN(price) && price.toFixed(2);

  if (!icon_url) {
    return null;
  }
  let name = item.name;
  const imgSrc = `${inventoryImageBaseUrl}${icon_url}`;
  const amount = count && count > 1 ? ` x ${count}` : '';

  if (name?.match(STAT_TRAK_PATTERN)) {
    name = name.replace(STAT_TRAK_PATTERN, '');
  }

  return (
    <Link className={styles.item} href={`/items/${assetid}`}>
      <Image src={imgSrc} priority alt={name} width={width} height={height} />
      {name && (
        <p style={{color: `#${name_color}`}} className={styles.describe}>
          {name + amount}
        </p>
      )}
      {formattedPrice && !isNaN(price) && <span className={styles.price}>{`${formattedPrice}$`}</span>}
    </Link>
  );
};
