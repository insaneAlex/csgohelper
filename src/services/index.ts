export {AWS_REGION, INVENTORY_TABLE} from './constants';
export {isClient} from './is-client';
export {storage} from './storage';
export {noop} from './noop';
export type {PriceType} from './types';
