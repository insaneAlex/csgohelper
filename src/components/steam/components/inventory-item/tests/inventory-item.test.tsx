import {InventoryItemType} from '@/src/services/steam-inventory';
import {render, screen} from '@testing-library/react';
import {InventoryItem} from '../inventory-item';
import {PriceOptions} from '../../../types';
import {Layout} from 'react-grid-layout';
import {NextRouter} from 'next/router';

jest.mock('../../../../../services', () => ({}));
const routerMock = {useRouter: () => ({push: jest.fn()})} as unknown as NextRouter;
const defaultProps = {router: routerMock, imgSize: {width: 1, height: 1}};
describe('InventoryItem', () => {
  const itemName = 'itmName';
  const item = {
    icon_url: 'icn',
    name_color: 'clr',
    name: itemName,
    assetid: 'id',
    exterior: 'f',
    prices: {[PriceOptions.WEEK]: {average: 2.004}}
  } as unknown as InventoryItemType & Layout;
  it('should render correctly', () => {
    const {container} = render(<InventoryItem {...defaultProps} item={item} />);
    expect(container).toMatchSnapshot();
  });
  describe('when no image', () => {
    it('should return null', () => {
      render(<InventoryItem {...defaultProps} item={{...item, icon_url: ''}} />);
      const link = screen.queryByRole('link');
      expect(link).not.toBeInTheDocument();
    });
  });
  describe('when stattrak item', () => {
    it('should render name without "StatTrak™ "', () => {
      render(<InventoryItem {...defaultProps} item={{...item, name: 'StatTrak™ PISTOL'}} />);
      const elem = screen.getByText('PISTOL');
      expect(elem).toBeInTheDocument();
    });
  });
  describe('when count is more than 1', () => {
    const count = 3;
    it('should render price prop', () => {
      render(<InventoryItem {...defaultProps} item={{...item, count, price: 3}} />);
      const price = `${(3).toFixed(2)}$`;
      const multiPrice = screen.getByText(price);
      expect(multiPrice).toBeInTheDocument();
    });
  });
});
