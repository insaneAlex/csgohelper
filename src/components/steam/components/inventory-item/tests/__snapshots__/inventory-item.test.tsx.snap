// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InventoryItem should render correctly 1`] = `
<div>
  <a
    class="item"
    href="/items/id"
  >
    <img
      alt="itmName"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="100"
      src="/_next/image?url=https%3A%2F%2Fsteamcommunity-a.akamaihd.net%2Feconomy%2Fimage%2Ficn&w=256&q=75"
      srcset="/_next/image?url=https%3A%2F%2Fsteamcommunity-a.akamaihd.net%2Feconomy%2Fimage%2Ficn&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fsteamcommunity-a.akamaihd.net%2Feconomy%2Fimage%2Ficn&w=256&q=75 2x"
      style="color: transparent;"
      width="100"
    />
    <p
      class="describe"
    >
      itmName
    </p>
    <span
      class="price"
    >
      2.00$
    </span>
  </a>
</div>
`;
